src/generator.cs

```diff
@@ -8000,9 +7999,10 @@ object GetDefaultValue (MethodInfo mi)
 		if (mi.ReturnType.IsEnum) {
 			if (def is string)
 				return def;
-			var name = mi.ReturnType.GetEnumName (def);
+			var name = Enum.GetName (mi.ReturnType, def);
 			if (string.IsNullOrEmpty (name)) {
-				return "(" + mi.ReturnType.FullName + ") " + def;
+				// The value could be negative so it need to be enclosed in parenthesis
+				return "(" + mi.ReturnType.FullName + ") (" + def + ")";
```

Comment:
There's combination of Mono bug and BGen bug. Mono fails to lookup the name from negative integer values. BGen generates invalid C# code when negative integers are present.

`DefaultValue (UIModalPresentationStyle.None)` used to take the `else` path here and end up as `return UIModalPresentationStyle.None!;` in the generated code.

With the Mono bug it doesn't find the name in the enumeration and it was converted to `return (UIKit.UIModalPresentationStyle) -1!` which is invalid C# code because of the negative number.

The new code generates `return (UIKit.UIModalPresentationStyle) (-1)!` which is obviously not ideal but at least it's valid C#.

The bug could also be triggered with current code by using `DefaultValue` attribute with a negative value not present in the enum.